frontend/src/components/Calendar/ClickableDateContainer.tsx

```diff
@@ -40,9 +42,18 @@ export const ClickableDateContainer = (props: {
 }) => {
   const dispatch = useDispatch()
   const { clickedDate, processedDates } = useSelector((state: RootState) => state.calendar)
+  const params = useParams<{ isEndDate: string }>()
 
   const DateContainerClickHandler = (newClickedDate: Date) => {
-    dispatch(setClickedDate(newClickedDate))
+    if (parseInt(params.isEndDate) === 0) {
+      dispatch(setClickedDate(newClickedDate))
+    } else {
+      const reccuringdate = newClickedDate.getTime() / 1000
+      if (Date.now() / 1000 > reccuringdate) {
+      } else {
+        dispatch(setBookingEndDate(reccuringdate))
+      }
+    }
     props.onDateClick(newClickedDate)
   }
```

Comment:
shouldn't be placing the codes here cos this component is supposed to be multi-purpose. instead, pass the custom function as a `props.onDateClick` param instead